chia/full_node/full_node_api.py

```diff
@@ -1279,7 +1279,7 @@ async def request_puzzle_solution(self, request: wallet_protocol.RequestPuzzleSo
         if header_hash is None:
             return reject_msg
 
-        block: Optional[FullBlock] = await self.full_node.block_store.get_full_block(header_hash)
+        block = await self.full_node.block_store.get_block_info(header_hash)
```

Comment:
the object returned here is one that implements the `BlockInfo` protocol. essentially a class that has the 3 members of `FullBlock` required by `get_block_generator()`. We don't want an actual full block here, because that's expensive to construct.

Reply:
Should we use the type annotation here `GeneratorBlockInfo`?

Reply:
maybe. I would probably be more tempted to annotate it with the `BlockInfo` protocol though. the concrete type is really an implementation detail